### PR TITLE
Add endpoint for saving token data

### PR DESCRIPTION
**Pull request by eccentricexit**

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -173,3 +189,36 @@ functions:
...
+          Action: ['KMS:Decrypt'],
+          Resource: '${self:custom.kmsSecrets.keyArn}',
+        }
+      - {
+          Effect: Allow,
+          Action: ['dynamodb:UpdateItem'],
+          Resource: 'arn:aws:dynamodb:us-east-2:547511976516:table/token',
+        }
+    documentation:
+      summary: 'Upload token data.'
```
Comment (epiqueras) [inline serverless.yml:211]: Upload => Update

File: serverless.yml [hunk 2]
```diff
@@ -173,3 +189,36 @@ functions:
...
+      description: 'Takes token data and stores it by content-address.'
```
Comment (epiqueras) [inline serverless.yml]: "Update token data for a content address (what is a content address?)."

File: src/tokens-on-trial/models/token.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "payload": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The name of the token."
+            },
+            "ticker": {
+              "type": "string",
+              "description": "The token ticker symbol."
```
Comment (epiqueras) [inline src/tokens-on-trial/models/token.json]: The token's ticker.

File: src/tokens-on-trial/models/token.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            },
+            "address": {
+              "type": "string",
+              "description": "The token Ethereum address."
+            },
+            "uri": {
```
Comment (epiqueras) [inline src/tokens-on-trial/models/token.json]: URI

File: src/tokens-on-trial/models/token.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        "payload": {
+          "type": "object",
+          "properties": {
+            "tokenID": {
+              "type": "string",
+              "description": "The tokenID for the uploaded token data."
+            },
+            "token": {
+              "type": "object",
+              "description": "The uploaded token data."
```
Comment (epiqueras) [inline src/tokens-on-trial/models/token.json]: Missing properties.

File: src/tokens-on-trial/token.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+const _web3 = require('../utils/web3')
+
+module.exports.patch = async (event, _context, callback) => {
```
Comment (epiqueras) [inline src/tokens-on-trial/token.js]: What you are doing is a PUT not a PATCH.

File: src/tokens-on-trial/token.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  callback(null, {
+    statusCode: 200,
+    headers: { 'Access-Control-Allow-Origin': '*' },
+    body: JSON.stringify({
+      payload: {
+        tokenID,
+        token: await dynamoDB.putItem({
+          Key: { tokenID: { S: tokenID } },
+          TableName: 'token',
+          UpdateExpression: keys.map(k => `SET ${k} = :_${k}`).join(', '),
```
Comment (epiqueras) [inline src/tokens-on-trial/token.js]: These are all part of Update/Patch Item. This won't work, see AWS docs.

File: yarn.lock [hunk 8]
```diff
@@ -5,12 +5,14 @@
 "@babel/code-frame@7.0.0-beta.44":
   version "7.0.0-beta.44"
   resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.0.0-beta.44.tgz#2a02643368de80916162be70865c97774f3adbd9"
+  integrity sha512-cuAuTTIQ9RqcFRJ/Y8PvTh+paepNcaGxwQwjIDRWPXmzzyAeCO4KqS9ikMvq0MCbRk6GlYKwfzStrcP3/jSL8g==
   dependencies:
```
Comment (epiqueras) [inline yarn.lock]: Make sure you have the latest version of yarn.

**Review by epiqueras — CHANGES_REQUESTED** [review]
Good start.

File: serverless.yml (hunk 2 above)
Comment (eccentricexit) [reply serverless.yml]: ~This is to mean that the key used to fetch token data will be the hash of that data.~
Nevermind, I misunderstood how dynamoDB queries would work.

**Comment by eccentricexit** [comment]
We can use cfs for storing token data, which means this is no longer required.

**Pull request closed without merging**